lib/rspec-puppet/adapters.rb

```diff
@@ -0,0 +1,107 @@
+module RSpec::Puppet
+  module Adapters
+
+    class Base
+      def setup_puppet(example_group)
+        settings_map.each do |puppet_setting, rspec_setting|
+          set_setting(example_group, puppet_setting, rspec_setting)
+        end
+      end
+
+      def set_setting(example_group, puppet_setting, rspec_setting)
+        if example_group.respond_to?(rspec_setting)
+          value = example_group.send(rspec_setting)
+        else
+          value = RSpec.configuration.send(rspec_setting)
+        end
+        begin
+          Puppet[puppet_setting] = value
+        rescue ArgumentError
+          # TODO: this silently swallows errors when applying settings that the current
```

Comment:
It's not the file itself that's private, it's actually the entire project: https://github.com/rspec/rspec-support/blob/master/README.md#rspecsupport . It's also a new dependency; apparently rspec-core doesn't have a hard dependency on it - or at least the version I have installed.

Anywho. Happy to use that if you like, but let's bounce this conversation over to the ticket :)